pkg/mimirpb/compat.go

```diff
@@ -593,20 +665,32 @@ loop:
 	return numLabels, true
 }
 
-// Generics
 type GenericSamplePair interface {
```

Comment:
I don't think this is used yet and also needs #4191 , so I'd just remove the whole GenericSamplePair code starting from line 668(new) for now.

Reply:
`GenericSamplePair` is already in the `main` branch, so I don't think removing and adding back is worth it. #4191 can be directly done in main branch now, isn't it?

Reply:
true, let's just keep it in mind